x-pack/legacy/plugins/advanced_ui_actions/public/customize_time_range_modal.tsx

```diff
@@ -0,0 +1,176 @@
+  };
+
+  cancel = () => {
+    this.props.onClose();
+  };
+
+  onInheritToggle = () => {
+    this.setState(prevState => ({
+      inheritTimeRange: !prevState.inheritTimeRange,
+    }));
+  };
+
+  addToPanel = () => {
+    const { embeddable } = this.props;
+
+    embeddable.updateInput({ timeRange: this.state.timeRange });
+
+    this.props.onClose();
+  };
+
+  inheritFromParent = () => {
+    const { embeddable } = this.props;
+    const parent = embeddable.parent;
+    const parentPanels = parent!.getInput().panels;
+
+    // Remove any explicit input to this child from the parent.
+    parent!.updateInput({
+      panels: {
+        ...parentPanels,
+        [embeddable.id]: {
+          ...parentPanels[embeddable.id],
+          explicitInput: {
+            ...parentPanels[embeddable.id].explicitInput,
+            timeRange: undefined,
+          },
+        },
+      },
+    });
+
+    this.props.onClose();
+  };
+
+  public render() {
+    return (
+      <React.Fragment>
+        <EuiModalHeader>
+          <EuiModalHeaderTitle data-test-subj="customizePanelTitle">
+            {i18n.translate('xpack.advancedUiActions.customizeTimeRange.modal.headerTitle', {
+              defaultMessage: 'Customize panel time range',
+            })}
+          </EuiModalHeaderTitle>
+        </EuiModalHeader>
+
+        <EuiModalBody>
+          {' '}
+          <EuiFormRow
+            label={i18n.translate(
+              'xpack.advancedUiActions.customizePanelTimeRange.modal.optionsMenuForm.panelTitleFormRowLabel',
+              {
+                defaultMessage: 'Time range',
+              }
+            )}
+          >
+            <EuiSuperDatePicker
+              start={this.state.timeRange ? this.state.timeRange.from : undefined}
+              end={this.state.timeRange ? this.state.timeRange.to : undefined}
+              isPaused={false}
+              onTimeChange={this.onTimeChange}
+              showUpdateButton={false}
+              dateFormat={this.props.dateFormat}
+            />
+          </EuiFormRow>
+        </EuiModalBody>
+        <EuiModalFooter>
+          <EuiFlexGroup justifyContent="spaceBetween">
+            <EuiFlexItem>
+              <EuiFlexGroup gutterSize="s">
+                <EuiButtonEmpty
+                  onClick={this.inheritFromParent}
+                  color="danger"
+                  data-test-subj="removePerPanelTimeRangeButton"
+                  disabled={this.state.inheritTimeRange}
+                >
+                  {' '}
+                  {i18n.translate(
+                    'xpack.advancedUiActions.customizePanelTimeRange.modal.removeButtonTitle',
+                    {
+                      defaultMessage: 'Remove',
+                    }
+                  )}
+                </EuiButtonEmpty>
+              </EuiFlexGroup>
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <EuiFlexGroup gutterSize="s">
```

Comment:
One too may `EuiFlexGroup`s here. Remove this one.

`EuiFlexItem` should be the closest parent to the actual contents.